### ISSUE TITLE
Looking for a maintainer of the SASS version

### ISSUE THREAD

**Issue by FezVrasta** [mdbootstrap/mdb-ui-kit#256]
Looks like @shockwork have not enough time to maintain the SASS files, if someone is willing to support me updating them to 0.2.0 please send a PR.

Thanks!

**Comment by ringofhealth** [comment]
Hey, I love what you have done. I'd love to help you out with this!

**Comment by FezVrasta** [comment]
Thank you, feel free to send PR about the SASS source. There was a SASS task that was removed early, you can find the gruntfile with the SASS tasks here:

https://github.com/FezVrasta/bootstrap-material-design/blob/2bbe921e5e73483aa4270b7afa8ea5fa6fe61753/Gruntfile.js

Feel free to re-implement the sass tasks following the new gruntfile style if you need them.

**Comment by dyaa** [comment]
Any updates !

**Comment by anuragrath** [comment]
Is the SCSS version maintained somewhere?

**Comment by FezVrasta** [comment]
No

**Comment by Soulou** [comment]
@FezVrasta I am also interested in the SASS support, what work should be done concretly, could you give some guidelines?

**Comment by blackfyre** [comment]
I've been doing some digging in the project, and a good start would to recreate the mixins from the LESS version.
The other modifications required are more or less just semantic ones.

**Comment by FezVrasta** [comment]
Yes it would be a good start.

**Comment by josephmilla** [comment]
Any update on this? I'd really like to have SASS support for this project.

**Comment by GradyD** [comment]
Not at this time but you are welcome to take ownership :D

**Comment by cerdiogenes** [comment]
I'm trying to address this, but I'm having difficult to convert _mixins-fullpalette.less to scss. Can someone give me a light in how to convert the nested mixins?

**Comment by karolyi** [comment]
+1, because of the lack of SASS version, paper in https://github.com/thomaspark/bootswatch won over this theme.

**Comment by rosskevin** [comment]
bootstrap 4 is switching to sass (which I'm already using).  Will this project switch to sass for 4.x as well?

**Comment by FezVrasta** [comment]
Yup

**Comment by rosskevin** [comment]
Any thoughts on making the _full_ switch now on a branch?  Maintaining a conversion process could be painful if the end goal is just to make the switch.  

If this is the case, I could take a shot at a conversion, but I don't want to be trying to maintain existing `less` fixes in the sass code.

**Comment by FezVrasta** [comment]
My idea is to write from scratch using bs4 as base

**Comment by rosskevin** [comment]
Got it, I'm planning to be on bs3 for the next several months at least (until bs4 and the related projects I use are worked out).  I'll take a shot at a conversion and see what happens.

**Comment by rosskevin** [comment]
To all watchers, I was able to wire up a grunt task to do the conversion here: https://github.com/rosskevin/bootstrap-material-design/tree/scss-conversion

It uses [grunt-less-to-sass](https://github.com/duvillierA/grunt-less-to-sass)  which is hopefully flexible enough to add some custom conversion rules to get things automated and correct.   I'll be trying this out, meanwhile I welcome pull requests on my fork.  Once stable, we'll submit a PR here.

**Comment by rosskevin** [comment]
The [`_mixins.scss`](https://github.com/rosskevin/bootstrap-material-design/blob/scss-conversion/sass/_mixins.scss) is definitely something that will need to be converted by hand.  Any help with that would be appreciated. 

Likewise, `_mixins-fullpalette.scss` will need the same treatment, but if I can get the fixes established for the first, this is much of the same.   Any help/guidance is appreciated.

``` ruby
$ grunt material:sass --trace

Running "sass:compile" (sass) task
Error: Undefined variable: "$material-color".
        on line 4 of sass/_mixins.scss, in `@content'
        from line 47 of sass/_mixins.scss, in `generic-variations'
        from line 3 of sass/_mixins.scss, in `variations'
        from line 20 of sass/_welljumbo.scss
        from line 41 of sass/material.scss
  Use --trace for backtrace.
Warning: Exited with error code 65 Use --force to continue.

Aborted due to warnings.
```

It appears that `@content` is static and not recognizing contextual variables.

**Comment by rosskevin** [comment]
the `generic-variations` mixins have been converted, but since there is no passing of functions in sass, and `@content` blocks aren't evaluated in the context of the variable defined at the time of inclusion, the code is a bit more static/less flexible.  I didn't look ahead at how much the `generic-variations` is used with a different content passed in such as in `_buttons.scss`.  We would have to implement a more static mixin for each of these uses...or...?  I'm not sure of a better solution.

Additionally, multi-line replacement doesn't appear to be working with `less-to-sass`, see https://github.com/duvillierA/grunt-less-to-sass/issues/6.

Perhaps @FezVrasta could take a look at [what I've had to do with `_mixins.scss`, `_mixins-fullpalette.scss`, and `_mixins-shared.scss`](https://github.com/rosskevin/bootstrap-material-design/tree/scss-conversion/sass).  There might be an alternate way to code this in `less` that  might make the conversion to `sass` a bit simpler.

**Comment by FezVrasta** [comment]
Sorry I'm not willing to spend time on this because I'll write from scratch everything even if this conversion is made.

I want to make a much more lightweight library, with NO javascript and without all the color variations.

**Comment by rosskevin** [comment]
For those subscribed, I have completed a (mostly) automated conversion to sass:
https://github.com/rosskevin/bootstrap-material-design/tree/scss-conversion

I've manually combed through a comparison of the output css files and it looks good.  

Grunt tasks are integrated to do the conversion at the same time a less release is created.

NOTE that the `_mixins-fullpalette.scss` requires further work.  While I won't need this, I have put in several `FIXME` notes describing the problem and the solution.

The only items _not_ automated had to do with the fact that sass does not allow pass-by-function.  This is noted by the use of the `generic-variations` mixin outside of the mixins files.  For example, some of the code in [`_mixins-shared.scss` such as `navbar-variations-content`](https://github.com/rosskevin/bootstrap-material-design/blob/scss-conversion/sass/_mixins-shared.scss#L85) will need to be manually updated to stay in-sync with the code in `_navbar.less`.  Luckily, the _only_ code that needs manual intervention is in `_mixins-shared.scss`.  I have noted in every `less` file where there is a potential impact e.g. [`_navbar.less`](https://github.com/rosskevin/bootstrap-material-design/blob/scss-conversion/less/_navbar.less#L169) so that we may more easily keep the codebase in-sync.   

I think this will actually be manageable.  I'll do some actual usage testing, then submit the PR.
